### PR TITLE
Dropdown Interactive Codemod Enhancements

### PR DESCRIPTION
**Pull request by zamoore**
### :pushpin: Summary

We received a report that the `dropdown-list-item-interactive` codemod didn't work correctly on yielded `Interactive` components that are rendered conditionally.

Example:

```
{{!
  Copyright (c) HashiCorp, Inc.
  SPDX-License-Identifier: MPL-2.0
}}
<Hds::Dropdown as |dd|>
  <dd.Title @text='Title text' />
  <dd.Interactive @href='#' @text={{or @text (t "components.generate-service-principal-key-button.default-text")}} />
</Hds::Dropdown>
```

The `Interactive` component was ignored because it was nested in another block.

### :hammer_and_wrench: Detailed description

- Changed the logic in the codemod to recursively traverse blocks in order to reach nested children
- Added another test case
- Added some clarifying comments

### :link: External links

Jira ticket: [HDS-4151](https://hashicorp.atlassian.net/browse/HDS-4151)

***

### 👀 Component checklist

- [ ] Percy was checked for any visual regression
- [ ] A changelog entry was added via [Changesets](https://github.com/changesets/changesets) if needed (see [templates here](https://hashicorp.atlassian.net/wiki/spaces/HDS/pages/3243114706/Changelog+authoring+best+practices#Templates))

:speech_balloon: Please consider using [conventional comments](https://conventionalcomments.org/) when reviewing this PR.

[HDS-4151]: https://hashicorp.atlassian.net/browse/HDS-4151?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

### CODE REVIEW

**Review by alex-ju — DISMISSED** [review]
Looking good! The only thing missing was processing the alternate part (what's inside the {{else}} block). I pushed a commit to cover that case as well (tried leaving suggestions, but it was quite tricky).

**Comment by zamoore** [comment]
> Looking good! The only thing missing was processing the alternate part (what's inside the {{else}} block). I pushed a commit to cover that case as well (tried leaving suggestions, but it was quite tricky).

Oh, good catch on that. Thanks.

**Comment by alex-ju** [comment]
> > Looking good! The only thing missing was processing the alternate part (what's inside the {{else}} block). I pushed a commit to cover that case as well (tried leaving suggestions, but it was quite tricky).
> 
> Oh, good catch on that. Thanks. I think pushing that code might have invalidated your review though with these new rules.

indeed, we need a final arbiter

**Review by shleewhite — COMMENTED** [review]
LGTM! 2 super small things: 
* this should probably have a changeset
* when I was looking into codemods, I noticed this one isn't listed in the [README](https://github.com/hashicorp/design-system/blob/main/packages/codemods/README.md#v4)

**Comment by alex-ju** [comment]
> LGTM! 2 super small things:
> 
> * this should probably have a changeset
> * when I was looking into codemods, I noticed this one isn't listed in the [README](https://github.com/hashicorp/design-system/blob/main/packages/codemods/README.md#v4)

good call! we should add it to the package readme and have a changelog entry for this patch

**Comment by zamoore** [comment]
Good call @shleewhite - I forgot we were doing changesets on codemods as well.

**Review by alex-ju — DISMISSED** [review]
Thanks for updating the changelog and docs. I made a small suggestion to the changelog but can be operated in the release PR as well. I'd wait for another approval since I became a contributor to this PR.

File: .changeset/lucky-hotels-sell.md
Comment (zamoore) [reply .changeset/lucky-hotels-sell.md]: LGTM, thanks for the code suggestion

**Review by alex-ju — APPROVED** [review]

**Review by shleewhite — APPROVED** [review]
🚢

File: packages/codemods/transforms/v4/dropdown-list-item-interactive/__testfixtures__/inside-conditional.input.hbs
Comment (zamoore) [reply packages/codemods/transforms/v4/dropdown-list-item-interactive/__testfixtures__/inside-conditional.input.hbs:8]: I think prettier is configured in the codemod package to use single quotes. My editor keeps converting them when I go double.

**Pull request merged**
